### PR TITLE
Use Paper's new ServerListPingEvent methods if present

### LINKED ISSUE #1419: Vanish doesn't remove players from server list ping in Paper 1.12
Pretty self-explanatory. When players enter vanish mode, EssentialsX would remove their name from the serverlist ping and decrease the player count returned to a query or game client pinging the server. However, this is no longer the case in EssentialsX 2.0.1-b501 (although I have a feeling that this occured when switching to 1.12 rather than by updating EssentialsX) and game clients or queries pinging the server will get the correct (visible+vanished) player count as well as the full list of player names.

EssentialsX version: 2.0.1-b501
Server software: git-Paper-1167 (MC: 1.12)

### PR DESCRIPTION
**Pull request by mdcfe**
Fixes #1419 - as of the Paper patch that added the new methods, there's now no way to change the server list player count on Paper 1.12+, but this PR fixes the names of vanished players showing up in the server list.

Tested against Spigot 1.12.1 and Paper 1.12.1, latest builds as of submitting PR.

### CODE REVIEW

File: Essentials/src/com/earth2me/essentials/EssentialsServerListener.java [hunk 1]
```diff
@@ -1,38 +1,76 @@
...
+import java.lang.reflect.Method;
 import java.util.Iterator;
+import java.util.List;
+import java.util.logging.Level;
 
 
 public class EssentialsServerListener implements Listener {
     private final transient IEssentials ess;
     private boolean errorLogged = false;
+    private Boolean isPaperSample;
```
Comment (SupaHam) [inline Essentials/src/com/earth2me/essentials/EssentialsServerListener.java]: primitive boolean

File: Essentials/src/com/earth2me/essentials/EssentialsServerListener.java [hunk 2]
```diff
@@ -1,38 +1,76 @@
...
     @EventHandler(priority = EventPriority.LOWEST)
     public void onServerListPing(final ServerListPingEvent event) {
-        try {
-            Iterator<Player> iterator = event.iterator();
-            while (iterator.hasNext()) {
-                Player player = iterator.next();
-                if (ess.getUser(player).isVanished()) {
-                    iterator.remove();
+        if (isPaperSample) {
+            // ess.getLogger().info("Paper sample");
```
Comment (SupaHam) [inline Essentials/src/com/earth2me/essentials/EssentialsServerListener.java]: boop

**Review by SupaHam — CHANGES_REQUESTED** [review]

File: Essentials/src/com/earth2me/essentials/EssentialsServerListener.java (hunk 1 above)
Comment (mdcfe) [reply Essentials/src/com/earth2me/essentials/EssentialsServerListener.java]: Leftover of previous messy implementation, will change now

**Comment by mdcfe** [comment]
@SupaHam Fixed

**Pull request merged**